src/Stratis.Bitcoin.Features.BlockStore/BlockStoreSignaled.cs

```diff
@@ -104,17 +116,54 @@ protected override void OnNextCore(Block block)
             this.logger.LogTrace("(-)");
         }
 
+        /// <summary>Initializes the <see cref="BlockStoreSignaled"/>.</summary>
+        public void Initialize()
+        {
+            this.batchTimer.Elapsed += async (sender, args) => { await this.OnBatchTimerRunsOutAsync().ConfigureAwait(false); };
+
+            this.blocksToAnnounce = new AsyncQueue<ChainedBlock>(async (item, cancellation) =>
+            {
+                using (await this.asyncLock.LockAsync(cancellation).ConfigureAwait(false))
+                {
+                    this.localBatch.Add(item);
+
+                    // Send batch right away if tip, if not and the timer haven't been started already- start the timer.
+                    if (item == this.chain.Tip)
+                    {
+                        this.batchTimer.Stop();
```

Comment:
Ah the timer is so that you can stop it from pinging, hmm..
This could easily be done with a flag.

Actually why do we need this?
Can we not just have an async loop that broadcast to peers if there are items in `this.localBatch` and if the condition `item == this.chain.Tip` we just not add the block to `this.localBatch.Add(item);` and broadcast it immediately?

Reply:
>we just not add the block to this.localBatch.Add(item); and broadcast it immediately?

we can't broadcast a single item if there are some items in the batch. peer will just discard the received block if it's based on a prev block that peer don't have. 

>Can we not just have an async loop that broadcast to peers 

You mean having an async loop with really small interval? That would be active waiting and cause huge resource consumption.

Reply:
>we can't broadcast a single item if there are some items in the batch

Ah yes correct then if there are items in the batch then just add to the batch and don't broadcast.

>You mean having an async loop with really small interval

The interval matters much less if we are not at tip right?
And we solved the tip problem already.
Agree though its a bigger resource consumption but its still very minor.

Reply:
I do however see the benefits of a stoppable timer. hmm.. 🤔

Reply:
>The interval matters much less if we are not at tip right?

Sure. But we really don't want to miss the moment (or delay it) when we finally hit the tip. And after we're synced we need to maintain low delays for every new block.

That was the original reason of making this PR- instead of just reducing the interval from 1 sec to smth smaller we are implementing the callbacks so we take action as soon as we have the block. 
So I'd really avoid using async loop to announce blocks. 
But async loop that will run once as a well logged solution for a timer is fine to me (even though it costs slightly more)

Reply:
>And after we're synced we need to maintain low delays for every new block

This you will anyway achieve.